core/templates/dev/head/pages/collection_editor/settings_tab/CollectionPermissionsCardDirective.js

```diff
@@ -22,10 +22,26 @@ oppia.directive('collectionPermissionsCard', [function() {
     restrict: 'E',
     templateUrl: 'inline/collection_permissions_card_directive',
     controller: [
-      '$scope',
-      function($scope) {
-        $scope.ownerNames = GLOBALS.ownerNames;
-        $scope.isPrivate = GLOBALS.isPrivate;
+      '$scope', 'CollectionRightsObjectFactory',
+      'CollectionRightsBackendApiService',
+      function($scope, CollectionRightsObjectFactory,
+        CollectionRightsBackendApiService) {
+        var collectionRightsObject =
+          CollectionRightsObjectFactory.createEmptyCollectionRights();
+        var refreshPermissionsCard = function() {
+          CollectionRightsBackendApiService.getCollectionRights(
+            GLOBALS.collectionId).then(function(data) {
+            $scope.ownerNames = data.owner_names;
```

Comment:
In fact -- if, here, you maintain a reference to the collection rights object, and then something somewhere else changes a property on that object, would the referenced property value be updated here as well? If so, that might be the best, since you don't need to have an additional $watch.